prow/plugins/size/pr.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package size
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/test-infra/prow/github"
+)
+
+// Strict subset of *github.Client methods.
+type githubClient interface {
+	AddLabel(owner, repo string, number int, label string) error
+	RemoveLabel(owner, repo string, number int, label string) error
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+	GetFile(org, repo, filepath, commit string) ([]byte, error)
+	GetPullRequestChanges(org, repo string, number int) ([]github.PullRequestChange, error)
+}
+
+func handlePR(gc githubClient, pe github.PullRequestEvent) error {
+	// These are the only two actions indicating the code diffs may have changed.
+	if pe.Action != "opened" && pe.Action != "synchronize" {
+		return nil
+	}
+
+	var (
+		owner = pe.PullRequest.Base.Repo.Owner.Login
+		repo  = pe.PullRequest.Base.Repo.Name
+		num   = pe.PullRequest.Number
+		sha   = pe.PullRequest.Base.SHA
+	)
+
+	// Ignore parseErrors, best effort despite invalid lines in config.
+	g, _, err := NewGenFilesGroup(gc, owner, repo, sha)
+	// Must fail early for io errors.
+	if err != nil {
+		return fmt.Errorf("io error while reading config: %s", err)
+	}
+
+	changes, err := gc.GetPullRequestChanges(owner, repo, num)
+	if err != nil {
+		return fmt.Errorf("GetPullRequestChanges error: %v", err)
+	}
+
+	var count int
+	for _, change := range changes {
+		if g.Match(change.Filename) {
+			continue
+		}
+
+		count += change.Additions + change.Deletions
+	}
+
+	labels, err := gc.GetIssueLabels(owner, repo, num)
```

Comment:
They do not appear to be contained [here](https://developer.github.com/v3/activity/events/types/#publicevent). even tho the structure is massive